src/main/java/teammates/storage/api/QueryWithParams.java

```diff
@@ -0,0 +1,40 @@
+package teammates.storage.api;
+
+import java.util.List;
+
+import javax.jdo.Query;
+
+/**
+ * Struct for bundling a Query together with parameters and expected result field.
```

Comment:
I haven't seen the word `Struct` being used in the in the Java world :-) `Data transfer object`?